### DISCUSSION TITLE
Remote server do not response util connection is closed.

### DISCUSSION THREAD

**Discussion by RyanCode** [FreeRDP/FreeRDP#8703] [Q&A] [answered]
I use `apache guacmole` which base on freerdp to connect to the remote desktop.In my test, if various events are triggered at high frequency, sometimes it will get stuck , and the remote machine will not send data packets(tcpdump will not recieve any packets).err log:`BIO_read returned a system error 110: Connection timed out`.I searched the history issue of freerdp and found no answer to this problem

**Comment by akallabeth** [comment, marked as answer]
@RyanCode ok, to sum up your description 'something went wrong'
sorry, please add a little more detail on what your setup is (configuration!), what you tested and what is going wrong, this can not be answered otherwise.

**Reply by RyanCode** [reply]
> @RyanCode ok, to sum up your description 'something went wrong' sorry, please add a little more detail on what your setup is (configuration!), what you tested and what is going wrong, this can not be answered otherwise.

I update freerdp to lasted version(2.10.0).It's resolved.thanks for your answer

**Reply by RyanCode** [reply]
> @RyanCode ok, to sum up your description 'something went wrong' sorry, please add a little more detail on what your setup is (configuration!), what you tested and what is going wrong, this can not be answered otherwise.

I will revert to the original freerdp version and check what the problem is

<img width="722" alt="image" src="https://user-images.githubusercontent.com/61410262/220065269-adff36ae-f74a-46d1-8f0b-50e646f4c417.png">

```log
[root@host ~]# DISPLAY=:0 /usr/bin/xfreerdp /buildconfig
This is FreeRDP version 2.1.1 (n/a)
Build configuration: BUILD_TESTING=OFF BUILTIN_CHANNELS=OFF HAVE_AIO_H=1 HAVE_EXECINFO_H=1 HAVE_FCNTL_H=1 HAVE_INTTYPES_H=1 HAVE_JOURNALD_H=TRUE HAVE_MATH_C99_LONG_DOUBLE=1 HAVE_POLL_H=1 HAVE_PTHREAD_MUTEX_TIMEDLOCK=ON HAVE_PTHREAD_MUTEX_TIMEDLOCK_LIB=1 HAVE_PTHREAD_MUTEX_TIMEDLOCK_SYMBOL= HAVE_SYSLOG_H=1 HAVE_SYS_EVENTFD_H=1 HAVE_SYS_FILIO_H= HAVE_SYS_MODEM_H= HAVE_SYS_SELECT_H=1 HAVE_SYS_SOCKIO_H= HAVE_SYS_STRTIO_H= HAVE_SYS_TIMERFD_H=1 HAVE_TM_GMTOFF=1 HAVE_UNISTD_H=1 HAVE_XI_TOUCH_CLASS=1 WITH_ALSA=ON WITH_CAIRO=OFF WITH_CCACHE=ON WITH_CHANNELS=ON WITH_CLANG_FORMAT=ON WITH_CLIENT=ON WITH_CLIENT_AVAILABLE=1 WITH_CLIENT_CHANNELS=ON WITH_CLIENT_CHANNELS_AVAILABLE=1 WITH_CLIENT_COMMON=ON WITH_CLIENT_INTERFACE=OFF WITH_CUPS=ON WITH_DEBUG_ALL=OFF WITH_DEBUG_CAPABILITIES=OFF WITH_DEBUG_CERTIFICATE=OFF WITH_DEBUG_CHANNELS=OFF WITH_DEBUG_CLIPRDR=OFF WITH_DEBUG_DVC=OFF WITH_DEBUG_KBD=OFF WITH_DEBUG_LICENSE=OFF WITH_DEBUG_MUTEX=OFF WITH_DEBUG_NEGO=OFF WITH_DEBUG_NLA=OFF WITH_DEBUG_NTLM=OFF WITH_DEBUG_RAIL=OFF WITH_DEBUG_RDP=OFF WITH_DEBUG_RDPDR=OFF WITH_DEBUG_RDPEI=OFF WITH_DEBUG_RDPGFX=OFF WITH_DEBUG_REDIR=OFF WITH_DEBUG_RFX=OFF WITH_DEBUG_RINGBUFFER=OFF WITH_DEBUG_SCARD=OFF WITH_DEBUG_SND=OFF WITH_DEBUG_SVC=OFF WITH_DEBUG_SYMBOLS=OFF WITH_DEBUG_THREADS=OFF WITH_DEBUG_TIMEZONE=OFF WITH_DEBUG_TRANSPORT=OFF WITH_DEBUG_TSG=OFF WITH_DEBUG_TSMF=OFF WITH_DEBUG_TSMF_AVAILABLE=0 WITH_DEBUG_WND=OFF WITH_DEBUG_X11=OFF WITH_DEBUG_X11_CLIPRDR=OFF WITH_DEBUG_X11_LOCAL_MOVESIZE=OFF WITH_DEBUG_XV=OFF WITH_DIRECTFB=OFF WITH_DSP_EXPERIMENTAL=OFF WITH_EVENTFD_READ_WRITE=1 WITH_FAAC=OFF WITH_FAAD2=OFF WITH_FFMPEG=OFF WITH_GFX_H264=OFF WITH_GPROF=OFF WITH_GSM=ON WITH_GSSAPI=OFF WITH_GSTREAMER_0_10=OFF WITH_GSTREAMER_1_0=ON WITH_ICU=ON WITH_IPP=OFF WITH_JPEG=ON WITH_LAME=OFF WITH_LIBRARY_VERSIONING=ON WITH_LIBSYSTEMD=ON WITH_MACAUDIO=OFF WITH_MACAUDIO_AVAILABLE=0 WITH_MANPAGES=ON WITH_MBEDTLS=OFF WITH_OPENCL=OFF WITH_OPENH264=OFF WITH_OPENSLES=OFF WITH_OPENSSL=ON WITH_OSS=ON WITH_PCSC=ON WITH_PROFILER=OFF WITH_PULSE=ON WITH_SAMPLE=OFF WITH_SANITIZE_ADDRESS=OFF WITH_SANITIZE_ADDRESS_AVAILABLE=1 WITH_SANITIZE_MEMORY=OFF WITH_SANITIZE_MEMORY_AVAILABLE=1 WITH_SANITIZE_THREAD=OFF WITH_SANITIZE_THREAD_AVAILABLE=1 WITH_SERVER=OFF WITH_SERVER_INTERFACE=OFF WITH_SHADOW_MAC=OFF WITH_SHADOW_X11=OFF WITH_SMARTCARD_INSPECT=OFF WITH_SOXR=OFF WITH_SSE2=ON WITH_SWSCALE=OFF WITH_THIRD_PARTY=OFF WITH_VAAPI=OFF WITH_VALGRIND_MEMCHECK=OFF WITH_VALGRIND_MEMCHECK_AVAILABLE=1 WITH_WAYLAND=ON WITH_WINPR_TOOLS=ON WITH_X11=ON WITH_X264=OFF WITH_XCURSOR=ON WITH_XEXT=ON WITH_XFIXES=ON WITH_XI=ON WITH_XINERAMA=ON WITH_XKBFILE=ON WITH_XRANDR=ON WITH_XRENDER=ON WITH_XSHM=ON WITH_XTEST=OFF WITH_XV=ON WITH_ZLIB=ON
Build type:          Release
CFLAGS:              -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic  -fPIC -Wall -Wno-unused-result -Wno-unused-but-set-variable -Wno-deprecated-declarations -fvisibility=hidden -Wimplicit-function-declaration -Wredundant-decls -DWINPR_DLL
Compiler:            GNU, 4.8.5
Target architecture: x64
```

**Reply by RyanCode** [reply]
> @RyanCode ok, to sum up your description 'something went wrong' sorry, please add a little more detail on what your setup is (configuration!), what you tested and what is going wrong, this can not be answered otherwise.

The clue is broken here
```log
BIO_read returned a system error 110: Connection timed out
transport_read_layer:freerdp_set_last_error_ex ERRCONNECT_CONNECT_TRANSPORT_FAILED [0x0002000D]
transport_check_fds: transport_read_pdu() - -1
transport_check_fds() - -1
```
Can you think of anything

**Reply by akallabeth** [reply]
that just means there was no data to read from a SSL socket, so not really helpful without context

**Reply by RyanCode** [reply]
> that just means there was no data to read from a SSL socket, so not really helpful without context

I found a phenomenon that whenever freerdp sends a large packet and is fragmented(It looks like an IP packet is divided into four 999 segments), the whole program will get stuck, that is, the client's read timeout,Is it possible that the server lost the packet or the client lost the response from the server? This should happen at the transport layer.There must be some exception message that causes the client to think that the tcp connection is abnormal



<img width="1413" alt="image" src="https://user-images.githubusercontent.com/61410262/221766734-15829ef3-ec2b-4608-9e30-0a8f6a83d9c3.png">


<img width="1379" alt="image" src="https://user-images.githubusercontent.com/61410262/221761635-d618eaf9-cacc-458b-af58-79c0c615119a.png">

**Reply by akallabeth** [reply]
@RyanCode before wasting time investigating a very old version you should probably just update the current client and check if the issue was already addressed.
(there have been 16 versions in between and I don´t remember every bug that was fixed ;))

**Reply by RyanCode** [reply]
I have upgraded the version to the latest,This problem will still arise

**Reply by akallabeth** [reply]
@RyanCode good. then 
1. which connection settings do you use?
2. which openssl version is in use?
3. what is the target RDP server? (windows version, `xrdp`, `ogon`, ...)
4. if possible, can you reproduce this with a direct connection (`xfreerdp`) too?

**Reply by RyanCode** [reply]
<img width="1395" alt="image" src="https://user-images.githubusercontent.com/61410262/221795591-8a131faf-13d9-45fd-9d25-d80df3f3008c.png">
It‘s out network problem,I test with wget,also send some TCP RST,But why is freerdp blocked and why not disconnect directly？

**Reply by akallabeth** [reply]
ok, so it is the network.
it blocks until the `SSL/TLS` connection times out, that is the reason. (just try a `wget` with a `https` url)
